### ISSUE TITLE
Updates  and Visual Studio Extension

### ISSUE THREAD

**Issue by brunobola** [volkanceylan/Serenity#4666]
Hello @volkanceylan, I've got to say that your project is very good for first starters, specifically in web development (.net/MVC/etc).

Have saying this when will you update your project to make easier for community and non-community users upgrade between versions without reading a lot of documentations.

Also I believe that your extension for visual studio don't compile out-of-the-box without some changes in the referenced files.

As always, great work and keep supporting the community with some changes to the base product to make it easier to people leap for the paid version.

Thanks, @brunobola.

**Comment by brunobola** [comment]
Any answer would be great, thanks.

**Comment by volkanceylan** [comment]
I dont think upgrade process is so complex and its not possible to do automated updates with NuGet packages in .Net core projects. Open to listen ideas and pull requests though.

**Comment by JohnRanger** [comment]
Hi @brunobola ,

I have recently upgraded a project from 2.9.24 to 3.9.6.

The upgrade process actually was absolutely straight forward - after I got it right :-)

**Don't try to do an inplace upgrade** - e.g. within your existing - old - project, calling update-package xxx. This didn't work well for me for such a big version step-up.

In the end, I created a completely new project, pointed it over to the old project's database and then I did import complete module folders incl. all their files from the old project into the new project.

And absolutely surprising (thanks again, @volkanceylan) the so upgraded project basically _just_ worked.

The only thing which I had to do was to do some "fixing" within the Serenity.css so some stuff looked again as it was looking in the "old" project - but that's just cosmetic.

With that I could do a complete upgrade of a medium complex project within less than an hour.

Hope this gives you a hint how to accomplish this.

P.S: This was with a "real" .NET SF-based app, not .NET core.

With kind regards,

John
